internal/resource-request-operator/broadcaster.go

```diff
@@ -147,36 +149,86 @@ func (b *Broadcaster) onNodeDelete(obj interface{}) {
+			if clusterID := newPod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+				klog.V(4).Infof("OnPodUpdate: Pod %s:%s passed ClusterID check. ClusterID = %s\n", newPod.Namespace, newPod.Name, clusterID)
+				currentPodsResources := b.readPodResources(clusterID)
+				// update the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
+				// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
+				updateResources(currentPodsResources, oldResources, newResources)
+				b.writePodsResources(clusterID, currentPodsResources)
+			}
+			// pod is becoming ready, same of onPodAdd case.
+		} else {
+			subResources(currentResources, newResources)
+			if clusterID := newPod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+				klog.V(4).Infof("OnPodUpdate: Pod %s:%s passed ClusterID check. ClusterID = %s\n", newPod.Namespace, newPod.Name, clusterID)
+				currentPodsResources := b.readPodResources(clusterID)
+				// add the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
+				// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
+				addResources(currentPodsResources, newResources)
+				b.writePodsResources(clusterID, currentPodsResources)
+			}
+		}
+		// pod is becoming not ready, same of onPodDelete case.
+	} else if pod.IsPodReady(oldPod) && !pod.IsPodReady(newPod) {
+		addResources(currentResources, newResources)
+		if clusterID := newPod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+			klog.V(4).Infof("OnPodUpdate: Pod %s:%s passed ClusterID check. ClusterID = %s\n", newPod.Namespace, newPod.Name, clusterID)
+			currentPodsResources := b.readPodResources(clusterID)
+			// sub the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
+			// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
+			subResources(currentPodsResources, oldResources)
+			b.writePodsResources(clusterID, currentPodsResources)
+		}
+	}
 	b.writeClusterResources(currentResources)
-	if clusterID := pod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
-		klog.V(4).Infof("OnPodDelete: Pod %s:%s passed ClusterID check. ClusterID = %s\n", pod.Namespace, pod.Name, clusterID)
-		currentPodsResources := b.readPodResources(clusterID)
-		subResources(currentPodsResources, podResources)
-		b.writePodsResources(clusterID, currentPodsResources)
+}
+
+func (b *Broadcaster) onPodDelete(obj interface{}) {
+	podDeleted := obj.(*corev1.Pod)
+	klog.V(4).Infof("OnPodDelete: Delete for pod %s:%s\n", podDeleted.Namespace, podDeleted.Name)
+	if pod.IsPodReady(podDeleted) {
+		podResources := extractPodResources(podDeleted)
+		currentResources := b.readClusterResources()
+		// this action is the complementary of the onPodAdd() one. For more details see at that function.
```

Comment:
Not really useful. I think you should explain what you mean and why it is important this "complementarity".